packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain.tsx

```diff
@@ -402,8 +411,39 @@ export function TransferPanelMain({
     return result
   }, [erc20L1Balances, erc20L2Balances, selectedToken])
 
-  const externalFrom = isConnectedToArbitrum ? l2.network : l1.network
-  const externalTo = isConnectedToArbitrum ? l1.network : l2.network
+  const externalFrom = useMemo(() => {
+    // if Arbitrum is L1...
+    if (isNetwork(l1.network.id).isArbitrum) {
+      // ...and we are connected to Arbitrum, then we deposit FROM L1 (L2 -> L3)
+      // otherwise we withdraw FROM L2 (L3 -> L2)
+      return isConnectedToArbitrum ? l1.network : l2.network
+    }
+    // if Ethereum is L1...
+    if (isNetwork(l1.network.id).isEthereum) {
+      // ...and we are connected to Arbitrum, then we withdraw FROM L2 (L2 -> L1)
+      // otherwise we deposit FROM L1 (L1 -> L2)
+      return isConnectedToArbitrum ? l2.network : l1.network
+    }
+    // if L3, we set FROM L2 (L3 -> L2) as default
+    return l2.network
+  }, [l1, l2])
+
+  const externalTo = useMemo(() => {
+    // if Arbitrum is L1...
+    if (isNetwork(l1.network.id).isArbitrum) {
+      // ...and we are connected to Arbitrum, then we deposit TO L2 (L2 -> L3)
+      // otherwise we withdraw TO L1 (L3 -> L2)
+      return isConnectedToArbitrum ? l2.network : l1.network
+    }
+    // if Ethereum is L1...
+    if (isNetwork(l1.network.id).isEthereum) {
+      // ...and we are connected to Arbitrum, then we withdraw TO L1 (L2 -> L1)
+      // otherwise we deposit TO L2 (L1 -> L2)
+      return isConnectedToArbitrum ? l1.network : l2.network
+    }
+    // if L3, we set TO L1 (L3 -> L2) as default
+    return l1.network
+  }, [l1, l2])
```

Comment:
I think this may be easier to understand if we did it together like this:

```
const [externalFrom, externalTo] = useMemo(...)
```

Also, let's make the L1-to-L2 or L2-to-L1 the base case, meaning what's executed if there are no early returns. It may be easier to understand that way.